cmd/legacy/main.go

```diff
@@ -252,3 +252,10 @@ func newKeyFromNetIPAddr(sip, dip netip.Addr, dport []byte, proto byte, isIPv4 b
 	}
 	return types.NewV6KeyStatic(sip.As16(), dip.As16(), dport, proto)
 }
+
+func ensureUnsigned(in int) uint64 {
```

Comment:
This forces the legacy metadata values smaller than 0 to 0. Crude, but for all intents and purposes, `-1 == 0`... :stuck_out_tongue:

Reply:
I'd rather take -1 == 0 than -1 == 2^64 -1 :). Looking at this, I distinctly remember a conversation with the person putting the code in place. The argument: _-1 will tell you that this shouldn't be taken at face value_.

I guessnot 🤓 .